### PR TITLE
v1.4 backports 2019-03-22

### PR DESCRIPTION
**Pull request by borkmann**
PRs:
- ~~#7414 *~~ (depends on #7349)
- #6984 *
- #7457 *

(*) conflicts

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7473)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-docs-please

**Comment by borkmann** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
all good from my commits

**Comment by borkmann** [comment]
timeout, retrying

**Comment by borkmann** [comment]
test-me-please

**Comment by ianvernon** [comment]
Panics were found in the logs:

```
13:46:55  ===================== TEST FAILED =====================
13:46:55  <Checks>
13:46:55  ⚠️  Found a "panic:" in logs
13:46:55  ⚠️  Found a "panic:" in logs
13:46:55  ⚠️  Found a "panic:" in logs
13:46:55  ⚠️  Found a "panic:" in logs
```

```
2019-03-22T13:31:19.060402673Z panic: runtime error: invalid memory address or nil pointer dereference
2019-03-22T13:31:19.060426929Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2739698]
2019-03-22T13:31:19.060430523Z 
2019-03-22T13:31:19.060433184Z goroutine 3207 [running]:
2019-03-22T13:31:19.060435784Z main.updateCNPNodeStatus(0xc0012aea00, 0xc001280101, 0x0, 0x0, 0x5, 0xc00129f230, 0xc000829ca0, 0xc000829ca0)
2019-03-22T13:31:19.060438513Z 	/go/src/github.com/cilium/cilium/daemon/k8s_watcher.go:1307 +0x468
2019-03-22T13:31:19.060441445Z main.cnpNodeStatusController(0x302bd80, 0xc000ca55a0, 0xc00121fcc0, 0x5, 0xc001289b30, 0x0, 0x0, 0x0, 0x0)
2019-03-22T13:31:19.060444117Z 	/go/src/github.com/cilium/cilium/daemon/k8s_watcher.go:1236 +0x5c1
2019-03-22T13:31:19.060446751Z main.(*Daemon).addCiliumNetworkPolicyV2.func1(0xbf1d56c9c291e707, 0x1650c3c05)
2019-03-22T13:31:19.060449359Z 	/go/src/github.com/cilium/cilium/daemon/k8s_watcher.go:1168 +0x60
2019-03-22T13:31:19.060451982Z github.com/cilium/cilium/pkg/controller.(*Controller).runController(0xc00128ed20)
2019-03-22T13:31:19.060454646Z 	/go/src/github.com/cilium/cilium/pkg/controller/controller.go:200 +0x47a
2019-03-22T13:31:19.060457182Z created by github.com/cilium/cilium/pkg/controller.(*Manager).UpdateController
2019-03-22T13:31:19.060459772Z 	/go/src/github.com/cilium/cilium/pkg/controller/manager.go:108 +0x576
```

File: daemon/k8s_watcher.go [hunk 1]
```diff
@@ -314,10 +314,7 @@ func (d *Daemon) EnableK8sWatcher(reSyncPeriod time.Duration) error {
 		return fmt.Errorf("Unsupported k8s version. Minimal supported version is %s", ciliumv2VerConstr.String())
 	}
 
-	ciliumNPClient, err = clientset.NewForConfig(restConfig)
-	if err != nil {
-		return fmt.Errorf("Unable to create cilium network policy client: %s", err)
-	}
+	ciliumNPClient := k8s.CiliumClient()
```
Comment (ianvernon) [inline daemon/k8s_watcher.go]: Changing this may be the cause of the panics while updating status.

Comment (borkmann) [reply daemon/k8s_watcher.go]: Thx, will check tomorrow if this can easily be resolved.

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-docs-please

**Comment by borkmann** [comment]
test-missed-k8s

**Comment by borkmann** [comment]
timeout getting vm, retrying

**Comment by borkmann** [comment]
test-missed-k8s

**Review by ianvernon — APPROVED** [review]

**Pull request merged**